core/lib/multivm/src/versions/vm_fast/vm.rs

```diff
@@ -842,15 +874,35 @@ impl<S: ReadStorage, T: Tracer> zksync_vm2::StorageInterface for World<S, T> {
     }
 }
 
+/// It may look like that an append-only cache for EVM bytecodes / `Program`s can lead to the following scenario:
+///
+/// 1. A transaction deploys an EVM bytecode with hash `H`, then reverts.
+/// 2. A following transaction in the same VM run queries a bytecode with hash `H` and gets it.
+///
+/// This would be incorrect behavior because bytecode deployments must be reverted along with transactions.
+///
+/// In reality, this cannot happen because both `decommit()` and `decommit_code()` calls perform storage-based checks
+/// before a decommit:
+///
+/// - `decommit_code()` is called from the `CodeOracle` system contract, which checks that the decommitted bytecode is known.
+/// - `decommit()` is called during far calls, which obtains address -> bytecode hash mapping beforehand.
+///
+/// Thus, if storage is reverted correctly, additional EVM bytecodes occupy the cache, but are unreachable.
+// FIXME: can it be used for DoS?
```

Comment:
- I don't quite like the reasoning above, but it seems to work for now at least.
- Regarding DoS, the hypothetical scenario is that the user can somehow cheaply add EVM bytecodes into the cache (perhaps eventually running out of gas), so that they persist for the entire VM run as deadweight. This could a problem only for batch processing (i.e., state keeper and VM runner-based components), not for the API server.

Reply:
I'm ok with this. If there is DoS it is not this code's fault. Adding a bytecode but not paying for it shouldn't be possible, even if reverted.